tests/baseline/tsfresh_features_test.py

```diff
@@ -0,0 +1,417 @@
+
+            def getKey(item):
+                return item[0]
+
+            sorted_feature_names = sorted(feature_names, key=getKey)
+            sorted_tsfresh_features = sorted(TSFRESH_FEATURES, key=getKey)
+
+            for nid, nname in sorted_feature_names:
+                new_entry = None
+                if int(nid) > max_known_id:
+                    new_entry = '    [%s, %s]' % (str(nnid), str(nname))
+                    fail_msg = '''
+
+###    NOTICE    ###
+
+If you adding/testing new features, you can create a new baseline from: %s
+
+Added the new baseline (Python 2 and 3 baselines ARE different) as either:
+Local path: %s/tsfresh-<NEW_VERSION>.py{2,3}.data.json.features.transposed.csv
+Repo path: tests/baseline/tsfresh-<NEW_VERSION>.py{2,3}.data.json.features.transposed.csv
+
+In the tsfresh_feature_names.py change the following:
+Local path: %s/tsfresh_feature_names.py
+Repo path: tests/baseline/tsfresh_feature_names.py
+
+- TSFRESH_BASELINE_VERSION to your <NEW_VERSION>
+- Add the new feature/s name to TSFRESH_FEATURES with a incremented id value
+
+New entry for tsfresh_feature_names.py:
+%s
+''' % (t_fname_out_fail, baseline_dir, baseline_dir, str(new_entry))
+                    if new_entry:
+                        shutil.move(t_fname_out, t_fname_out_fail)
+                    self.assertEqual(new_entry, None, msg=fail_msg)
+
+                for oid, oname in sorted_feature_names:
+                    if int(oid) == int(nid):
+                        if str(oname) != str(nname):
+                            soid = str(int(oid))
+                            snid = str(int(nid))
+                            soname = str(oname)
+                            snname = str(oname)
+                            fail_msg = '''
+
+###    ERROR    ###
+
+A baseline feature name for id %s has changed:
+    [%s, %s]
+Calculated feature name for id %s was:
+    [%s, %s]
+
+If you adding/testing new features, you can create a new baseline from: %s
+
+Added the new baseline (Python 2 and 3 baselines ARE different) as either:
+Local path: %s/tsfresh-<NEW_VERSION>.py{2,3}.data.json.features.transposed.csv
+Repo path: tests/baseline/tsfresh-<NEW_VERSION>.py{2,3}.data.json.features.transposed.csv
+
+In the tsfresh_feature_names.py change the following:
+Local path: %s/tsfresh_feature_names.py
+Repo path: tests/baseline/tsfresh_feature_names.py
+
+- TSFRESH_BASELINE_VERSION to your <NEW_VERSION>
+- Add the new feature/s name to TSFRESH_FEATURES with a incremented id value
+
+''' % (soid, soid, soname, soid, soid, snname, t_fname_out, baseline_dir, baseline_dir)
+                            if soname != snname:
+                                shutil.move(t_fname_out, t_fname_out_fail)
+                            self.assertEqual(soname, snname, msg=fail_msg)
+            feature_names_match = True
+        else:
+            feature_names_match = False
+            shutil.move(t_fname_out, t_fname_out_fail)
```

Comment:
@nils-braun I think the last commit is more in line with what you are describing
as the desired, it is just comparing the dataframes and I have removed the list
of features file and it is just going on the names in the baseline.

I thought it would be possible to carry the setUp df_t Dataframe through as
self.df_t but it returns, even when copying the df_t object to self.df_trans
so that to_csv is taken out of the equation, probably just some failing on my
part. 

```
>   ???
E   IndexError: index out of bounds
```

So the test is having to read the csvs into a Dataframes again, but that works,
but is not ideal it would be better to use self.df_t if possible.

While debugging the version issue #109, I ran the test on 0.1.2 with the 0.3.0 baseline and this is the reduced comment string blow up output.

```
E       AssertionError:
E       See the docs on how to update the baseline.
E       New local baseline: /opt/python_virtualenv/projects/tsfresh012-py2712/apps/tsfresh_test_tsfresh/tsfresh/tsfresh/examples/data/test_tsfresh_baseline_dataset/tsfresh-unknown-version.py2.data.json.features.transposed.csv
E
E       NOT in baseline   :: [['value__sum_of_reoccurring_values', '109822.0'], ['value__sample_entropy', '2.22646139752'], ['value__ratio_value_number_to_time_series_length', '0.9'], ['value__percentage_of_reoccurring_values_to_all_values', '0.183333333333'], ['value__percentage_of_reoccurring_datapoints_to_all_datapoints', '0.0925925925926'], ['value__approximate_entropy__m_2__r_0.9', '0.679727853081'], ['value__approximate_entropy__m_2__r_0.7', '0.770566235381'], ['value__approximate_entropy__m_2__r_0.5', '0.816629919728'], ['value__approximate_entropy__m_2__r_0.3', '0.556059428563'], ['value__approximate_entropy__m_2__r_0.1', '0.0768916250217']]
E
E       NOT in calculated :: []

tests/baseline/tsfresh_features_test.py:197: AssertionError
```

On the version issue, unfortunately for now the version has to be hardcoded into the test.

I was thinking it would be probably be possible to handle the py2 and py3
differences with the numpy assert_almost_equal method to allow for a single
baseline that would work on either, by using the assert_almost_equal the Python
version being used for testing does not match the version of Python on which the
baseline was created, however that gets a bit trickier, I reckon it is possible
but at a later date.

### But

I was intrigued how the Python 3.5.2 Travis job failed like that - https://travis-ci.org/blue-yonder/tsfresh/jobs/183891740
I obviously cannot replicate that, from the job output it is using the py3 baseline, that is confirmed by the presence of:

'value__cwt_coefficients__widths_(2, 5, 10, 20)__coeff_9__w_5', '6826.238621617837'

My 3.5.2 baseline calculation: '6826.238621617838'
Travis 3.5.2 calculation:      '6826.238621617837'

```
E       New local baseline: /home/travis/build/blue-yonder/tsfresh/tsfresh/examples/data/test_tsfresh_baseline_dataset/tsfresh-unknown-version.py3.data.json.features.transposed.csv
E       
E       NOT in baseline   :: [['value__cwt_coefficients__widths_(2, 5, 10, 20)__coeff_9__w_5', '6826.238621617838'],
...
...
E       NOT in calculated :: [['value__cwt_coefficients__widths_(2, 5, 10, 20)__coeff_9__w_5', '6826.238621617837'],
```

So this all needs to be wrapped up in assert_almost_equal?  There are probably
valid internal or process/algorithm related reasons for these values to be off
by minor margin, inconvenient as it may be.  Perhaps this test is a bad idea,
however I would like it work and perhaps be useful.  Open to ideas or closing
the PR if the overhead is not worth the upkeep.  Is this in any way a good idea? :)